src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/UnixNetworkInterface.cs

```diff
@@ -76,6 +72,7 @@ protected unsafe void ProcessIpv4Address(Interop.Sys.IpAddressInfo* addressInfo,
             IPAddress netMaskAddress = IPAddressUtil.GetIPAddressFromNativeInfo(netMask);
             AddAddress(ipAddress);
             _netMasks[ipAddress] = netMaskAddress;
+            _index = addressInfo->InterfaceIndex;
         }
```

Comment:
@tmds Can you add a test for this?

Reply:
I can add a test that loops over the interfaces and checks the Index property is not -1. The Index is not exposed, so it may need some reflection.
Without the changes in this PR, that test will fail on a machine with a tun interface. Probably the CI machines don't have such an interface, so that case wouldn't be checked in CI.
@Priya91 wdyt?

Reply:
Hmm, doesn't seem like a good test.

Reply:
How about loppback? That does not have HW address either @tmds

Reply:
@wfurt I had thought that too. `ProcessLinkLayerAddress` is called with a 6-byte zero address for loopback.

Reply:
that is interesting. ifconfig does not show any. But I could imagine system fakes ethernet MAC will all zeroes.